Add option to override per-VR character set usage

Enet4 opened the pull request:
Should resolve #675 by offering a way to ignore the fact that some VRs should always use the default character repertoire.

### Summary

- [parser] Add support for character set override logic
- [object] Extend charset_override option to collector and in-mem DICOM object

naterichman reviewed (commented):
Sorry if I'm missing something, but figured I would hop on a review in case it helped!

momostarsky commented:
@Enet4  ​​Hi, why hasn't this PR been merged yet? Is there anything blocking it? Does it need more testing?​

Enet4 commented:
I just didn't get to it yet. Let me assess and revise on Nathan's comment first, hopefully within the next few days.

Enet4 commented:
I rebased the branch and added UR as one of the VRs to use the default character set by default. Things seem to be in order. @momostarsky I noticed a deleted post here, are there still any issues?

Enet4 commented:
That sounds to offer good coverage already, thanks! If in the meantime you found what could be considered a bug, please file a new issue.
